backend/src/main/java/ai/giskard/domain/ApiKey.java

```diff
@@ -1,15 +1,19 @@
 package ai.giskard.domain;
 
+import ai.giskard.security.GalleryDatabaseOperationListener;
 import jakarta.persistence.*;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
+import lombok.Setter;
 import org.apache.commons.lang3.RandomStringUtils;
 
 import java.util.UUID;
 
 @Entity(name = "api_keys")
 @Getter
+@Setter
 @NoArgsConstructor
+@EntityListeners(GalleryDatabaseOperationListener.class)
```

Comment:
We should probably move the `@EntityListeners(GalleryDatabaseOperationListener.class)` annotation to the `AbstractAuditingEntity` and/or `BaseEntity` to avoid having to put it everywhere and forgot to add it for future entities